packages/uiweb/src/lib/components/supportChat/Modal.tsx

```diff
@@ -59,17 +59,32 @@ export const Modal: React.FC = () => {
     timestamp: undefined,
     icon: <HandWaveSvg fill={theme.btnColorPrimary}/>,
   };
-  const onScroll = () => {
+  const onScroll = async () => {
     if (listInnerRef.current) {
       const { scrollTop } = listInnerRef.current;
       if (scrollTop === 0) {
-        // This will be triggered after hitting the first element.
-        // pagination
-        getChatCall();
+        const content = listInnerRef.current;
+        const curScrollPos = content.scrollTop;
+        const oldScroll = content.scrollHeight - content.clientHeight;
+
+        await getChatCall();
```

Comment:
I think there is no need to await this function here

Reply:
@KlausMikhaelson it is fetching all the history chats from PushAPI and hence it's nice to make it an async call